### PR TITLE
Implemented MakeTransposable and tested simple case of GetStatesBatches

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

**Comment by danpovey** [comment]
You can merge when you think it's ready.

File: k2/csrc/ragged_ops.cu [hunk 1]
```diff
@@ -683,6 +683,101 @@ RaggedShape RemoveAxis(RaggedShape &src, int32_t axis) {
...
+        Eval(c, src_dim0, lambda_set_row_splits2);
+        ExclusiveSum(offsets, &offsets);
+        int32_t tot_size2 = src.TotSize(2);
+        axis2_shape.row_ids = Array1<int32_t>(c, tot_size2);
+        int32_t *ans_row_ids2_data = axis2_shape.row_ids.Data();
+        const int32_t *src_row_ids2_data = src.RowIds(2).Data();
+        auto lambda_set_row_ids2 = [=] __host__ __device__(int32_t idx012) {
+          int32_t src_idx01 = src_row_ids2_data[idx012];
+          int32_t src_idx0 = src_row_ids1_data[src_idx01];
+          ans_row_ids2_data[idx012] = src_idx01 + offsets_data[src_idx0];
```
[the remark this answers is not in the document]
Comment (qindazhu) [reply k2/csrc/ragged_ops.cu:770]: Oh, thanks, will update.

[the remark this answers is not in the document]
Comment (qindazhu) [reply k2/csrc/ragged_ops.cu]: Fixed.

**Comment by qindazhu** [comment]
Merging..even though still debug  the  complex version GetStatesBatches.

**Pull request merged**